system/lib/wasmfs/backends/ignore_case_backend.cpp

```diff
@@ -129,16 +212,21 @@ class IgnoreCaseBackend : public Backend {
   }
 
   std::shared_ptr<DataFile> createFile(mode_t mode) override {
-    return backend->createFile(mode);
+    return virtualize(backend->createFile(mode), this);
   }
 
   std::shared_ptr<Directory> createDirectory(mode_t mode) override {
-    return std::make_shared<IgnoreCaseDirectory>(backend->createDirectory(mode),
-                                                 this);
+    auto real = backend->createDirectory(mode);
+    // Inserts into the real backing directory won't work if it doesn't appear
+    // to be linked, so give it a parent.
+    // TODO: Break this reference cycle in a destructor somewhere.
```

Comment:
Should this be done in the current PR?

Reply:
Sure, I guess this is easy enough to add to the backend destructor.